client/web/src/settings/DynamicallyImportedMonacoSettingsEditor.tsx

```diff
@@ -91,18 +99,33 @@ export class DynamicallyImportedMonacoSettingsEditor extends React.PureComponent
     }
 
     public render(): JSX.Element | null {
-        const isDirty = this.isDirty
         const effectiveValue = this.effectiveValue
+
+        let saveToolbar: React.ReactElement | null = null
+        if (this.props.customSaveToolbar) {
+            const toolbarProps = this.props.customSaveToolbar.propsGenerator({
+                dirty: this.isDirty,
+                saving: this.props.saving,
+                onSave: this.onSave,
+                onDiscard: this.discard
+            })
+            saveToolbar = this.props.customSaveToolbar.saveToolbar(toolbarProps)
+        }
+
         return (
             <div className={this.props.className || ''}>
-                {this.props.canEdit && (
-                    <SaveToolbar
-                        dirty={isDirty}
-                        saving={this.props.saving}
-                        onSave={this.onSave}
-                        onDiscard={this.discard}
-                    />
-                )}
+                {
+                    this.props.canEdit &&
+                    (
+                        saveToolbar ||
+                        <SaveToolbar
+                            dirty={this.isDirty}
+                            saving={this.props.saving}
+                            onSave={this.onSave}
+                            onDiscard={this.discard}
+                        />
+                    )
+                }
```

Comment:
You could add an else branch above in l.113 so you dont need a second switch here :) but no blocker.
```suggestion
                        saveToolbar
                }
```